sdp/pyfftw_sdp.py

```diff
@@ -4,48 +4,86 @@
 
 class SLIDING_DOT_PRODUCT:
     # https://stackoverflow.com/a/30615425/2955541
-    def __init__(self):
-        self.m = 0
+    def __init__(self, max_shape=2**10):
```

Comment:
Should we remove `max_shape` from here, and get its value from `config.py` instead? 

```
def __init__(self):
    max_shape = config.PYFFTW_MAX_SHAPE
```